src/js/util/cookieFunctions.ts

```diff
@@ -0,0 +1,39 @@
+import Cookies from 'js-cookie';
+import { CookieOptions } from '../types/cookies';
+
+//set cookies function
+export const setCookies = (
+    name: string,
+    value: string,
+    durationInDays: number,
+    subDomains?: string[],
+    mainDomain?: string
+) => {
+    if (typeof location === 'undefined' || typeof document === 'undefined')
+        return;
+
+    const allDomains = [
+        ...(subDomains?.map((subD) => `${subD}.${mainDomain}`) || []),
+        mainDomain,
+    ];
+
+    if (allDomains.includes(document.location.hostname)) {
+        Cookies.set(name, value, {
+            expires: durationInDays,
+            domain: mainDomain,
+        });
+    } else {
+        Cookies.set(name, value, {
+            expires: durationInDays,
+        });
+    }
+};
+
+export const cookieOptionsToArray = (cookieOptions: CookieOptions) => {
+    const { analytics, marketing, functional } = cookieOptions;
+    let cookieValue = [];
+    if (functional) cookieValue.push('functional');
```

Comment:
Die CookieOptions komen als string regelmatig terug. Kun je daar ook een constante van maken.

Reply:
Wat bedoel je precies?

Reply:
Je kan bv een enum maken met die CookieGroups:

```
export enum CookieGroup {
    Functional = 'functional',
    Marketing = 'marketing',
    Analytics = 'analytics',
}
```

Die kan je dan zowel als type check gebruiken of als constate:
```
cookieValue.push(CookieGroup.Analytics)
```

Reply:
Yes!